notebooks/304-pytorch-post-training-quantization/304-pytorch-post-training-quantization.ipynb

```diff
@@ -0,0 +1,698 @@
+    "                progress.display(i)\n",
+    "\n",
+    "        print(' * Acc@1 {top1.avg:.3f} Acc@5 {top5.avg:.3f}'\n",
+    "              .format(top1=top1, top5=top5))\n",
+    "    return top1.avg"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1912cb42",
+   "metadata": {},
+   "source": [
+    "### Create and load original uncompressed model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3a92b300",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_model(model_path):\n",
+    "    model = models.resnet50(pretrained=True)\n",
+    "    # update the last FC layer for Tiny ImageNet number of classes\n",
+    "    NUM_CLASSES = 200\n",
+    "    model.fc = nn.Linear(in_features=2048, out_features=NUM_CLASSES, bias=True)\n",
+    "    model.to(device)\n",
+    "    if model_path.exists():\n",
+    "        checkpoint = torch.load(str(model_path), map_location='cpu')\n",
+    "        model.load_state_dict(checkpoint['state_dict'], strict=True)\n",
+    "    else:\n",
+    "        raise RuntimeEror('There is no checkpoint to load')\n",
+    "    return model\n",
+    "\n",
+    "\n",
+    "model = create_model(MODEL_DIR / fp32_checkpoint_filename)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4291ed2f",
+   "metadata": {},
+   "source": [
+    "### Create train and validation dataloaders "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "93213778",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_dataloaders(batch_size=128):\n",
+    "    train_dir = DATASET_DIR / 'train'\n",
+    "    val_dir = DATASET_DIR / 'val' / 'images'\n",
+    "    normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406],\n",
+    "                                     std=[0.229, 0.224, 0.225])\n",
+    "    train_dataset = datasets.ImageFolder(\n",
+    "        train_dir,\n",
+    "        transforms.Compose([\n",
+    "            transforms.Resize(IMAGE_SIZE),\n",
+    "            transforms.ToTensor(),\n",
+    "            normalize,\n",
+    "        ]))\n",
+    "    val_dataset = datasets.ImageFolder(\n",
+    "        val_dir,\n",
+    "        transforms.Compose([\n",
+    "            transforms.Resize(IMAGE_SIZE),\n",
+    "            transforms.ToTensor(),\n",
+    "            normalize\n",
+    "        ]))\n",
+    "\n",
+    "    train_loader = torch.utils.data.DataLoader(train_dataset, batch_size=batch_size, shuffle=True,\n",
+    "                                               num_workers=4, pin_memory=True, sampler=None)\n",
+    "    \n",
+    "    val_loader = torch.utils.data.DataLoader(val_dataset, batch_size=batch_size, shuffle=False,\n",
+    "                                             num_workers=4, pin_memory=True)\n",
+    "    return train_loader, val_loader\n",
+    "\n",
+    "\n",
+    "train_loader, val_loader = create_dataloaders()"
+   ]
+  },
+  {
```

Comment:
Since we are now going towards the `I. II. III.` steps, some plain text is required to visually separate the preparation part from the part where all of the actual magic happens. Add a paragraph here to the tune of: `Now that the preparation of the training and validation pipelines and the model files within this notebook is done, we will walk through the major steps of performing actual post-training quantization with NNCF.`

Reply:
Done